### ISSUE TITLE
Onion Services v3 support

### ISSUE THREAD

**Issue by cypherbits** [s-rah/onionscan#157]
v3 Hidden Services cannot be scanned.

**Comment by lorenzoromani1983** [comment]
You have to add the following lines in utils/validation:

	if len(identifier) >= 62 && strings.HasSuffix(identifier, ".onion") {
		matched, _ := regexp.MatchString(`(^|\.)[a-z2-7]{56}\.onion$`, identifier)

this will not trigger the unrecognised onion error.

After editing the file, repete the install procedure and you will be ok with v3 addresses.

this worked for me.

**Comment by cypherbits** [comment]
@lorenzoromani1983 yes yes, I already did that in my fork and even added and changed many other things. But I opened the issue here so it can get officially fixed sometime.

**Comment by Finleyyy** [comment]
> 
> 
> You have to add the following lines in utils/validation:
> 
> ```
> if len(identifier) >= 62 && strings.HasSuffix(identifier, ".onion") {
> 	matched, _ := regexp.MatchString(`(^|\.)[a-z2-7]{56}\.onion$`, identifier)
> ```
> 
> this will not trigger the unrecognised onion error.
> 
> After editing the file, repete the install procedure and you will be ok with v3 addresses.
> 
> this worked for me.

First, is it ~/go/src/github.com/s-rah/onionscan/utils/validation.go? If yes, where do I add those lines? Because it doesn't work for me. My current validation.go is;
`package utils

import (
	"regexp"
	"strings"
)

func IsOnion(identifier string) bool {
	// TODO: At some point we will want to support i2p
	if len(identifier) >= 62 && strings.HasSuffix(identifier, ".onion") {
	matched, _ := regexp.MatchString(`(^|\.)[a-z2-7]{56}\.onion$`, identifier)
		return matched
	}
	return false
}
`

**Comment by brg123** [comment]
I tried just to copy it, but i dont know where it has to be placed. Could anyone show the full validation.go file source?
After editing i did go install github.com/s-rah/onionscan or was something meant with repeate the install procedere?

Thanks a lot

**Comment by q2dg** [comment]
Well, it seems this project is abandoned.
Sad

**Comment by nicomitor** [comment]
Well in fact the fix is quite easy ;-)
You have to add the following lines in utils/validation:

if len(identifier) >= 62 && strings.HasSuffix(identifier, ".onion") {
	matched, _ := regexp.MatchString(`(^|\.)[a-z2-7]{56}\.onion$`, identifier)       <<<replace 16 in the original file with 56, which is the number of characters of a v3 hostname.  That's it.
